tests/upgrades/test_host.py

```diff
@@ -0,0 +1,152 @@
+                'locations.resources.assigned': [loc.name]})
+        gce_cr = entities.AbstractComputeResource().search(
+            query={'search': 'name={}'.format(cr_name)})[0]
+        gce_img = entities.Image(
+            architecture=self.arch, compute_resource=gce_cr, name='autoupgrade_gce_img',
+            operatingsystem=self.os, username='gceautou', uuid=self.img_uuid).create()
+        create_dict({self.__class__.__name__: {
+            'org': org.name, 'loc': loc.name, 'cr_name': cr_name}})
+        assert gce_cr.name == cr_name
+        assert gce_img.name == 'autoupgrade_gce_img'
+
+    @post_upgrade(depend_on=test_pre_create_gce_cr_and_host)
+    def test_post_create_gce_cr_and_host(self):
+        """
+        """
+        self.hostname = gen_string('alpha')
+        preentities = get_entity_data(self.__class__.__name__)
+        gce_cr = entities.GCEComputeResource().search(
+            query={'search': 'name={}'.format(preentities['cr_name'])})[0]
+        org = entities.Organization().search(
+            query={'search': 'name={}'.format(preentities['org'])})[0]
+        loc = entities.Location().search(
+            query={'search': 'name={}'.format(preentities['loc'])})[0]
+        compute_attrs = {
+            'machine_type': 'g1-small', 'network': 'default', 'associate_external_ip': True,
+            'volumes_attributes': {'0': {'size_gb': '10'}}, 'image_id': self.img_uuid}
+        # Host Provisioning Tests
+        try:
+            skip_yum_update_during_provisioning(template='Kickstart default finish')
+            gce_hst = entities.Host(
+                name=self.hostname,
+                organization=org,
+                location=loc,
+                root_pass=gen_string('alphanumeric'),
+                architecture=self.arch,
+                compute_resource=gce_cr,
+                domain=entities.Domain().search(
+                    query={'search': 'name={}'.format(self.domain_name)})[0],
+                compute_attributes=compute_attrs,
+                operatingsystem=self.os,
+                provision_method='image'
+            ).create()
+        finally:
+            skip_yum_update_during_provisioning(template='Kickstart default finish', reverse=True)
+        wait_for(
+            lambda: entities.Host().search(
+                query={'search': 'name={}.{}'.format(
+                    self.hostname, self.domain_name)
+                })[0].build_status_label == 'Installed',
+            timeout=400,
+            delay=15,
+            silent_failure=True,
+            handle_exception=True
+        )
+        assert gce_hst.name == '{}.{}'.format(self.hostname, self.domain_name).lower()
+        gce_hst = entities.Host(id=gce_hst.id).read()
+        assert gce_hst.build_status_label == 'Installed'
+        # CR Manipulation Tests
+        newgce_name = gen_string('alpha')
+        gce_cr.name = newgce_name
+        gce_cr.zone = 'europe-west4-a'
```

Comment:
Use from robottelo.constants import VALID_GCE_ZONES

Reply:
With this update, I am checking if the 'update' functionality works fine not 'GCE zones'. For GCE Zones, the robottelo test will take care of it. Hence hardcoded this value.

Reply:
If VALID_GCE_ZONES is used, then we don't need to maintain the test case if zone value is changed in future.

Reply:
Done